02-express-tutorial/app.js

```diff
@@ -1 +1,55 @@
-console.log('Express Tutorial')
+const express = require('express')
+const app = express();
+const{ product } = require('./data')
+
+app.use(express.static("./public"))
+
+app.get('/api/v1/test',(req,res) => {
+    res.json({ message: "It worked!" });
+});
+
+app.get('/api/v1/products',(req,res) => {
+    res.json({message: "product"}) 
+    
+})
+
+app.get('/api/v1/products/:productID',(req,res) => {
+   // res.json(req.params)
+   const idToFind = parseInt(req.params.productID); 
+   const product = products.find((p) => p.id === idToFind);
+
+    if(product) {
+        res.status(404).json ({ message: "That product was not found" })
+    } else {
+       res.json(product);
+    }
```

Comment:
Nitpick (doesn't require changes):
I like to avoid if/else statements for 2 conditions and instead, either use early returns for errors/invalid responses:
```js
if (!product) {
  return res.status(404).json({ message: "That product was not found." });
}

return res.json(product);
```
OR, use a ternary operator:
```js
return !product
  ? res.status(404).json({ message: "That product was not found." }) 
  : res.json(product);
```